shotover-proxy/tests/cassandra_int_tests/routing.rs

```diff
@@ -1,93 +1,336 @@
+                    ],
+                )
+                .await;
+            assert_eq!(shotover_hit, cassandra_hit);
+        }
+
+        for key in 0..10 {
+            let shotover_hit = shotover
+                .execute_prepared_coordinator_node(
+                    &prepared_select,
+                    &[ResultValue::Int(key), ResultValue::Int(key)],
+                )
+                .await;
+            let cassandra_hit = cassandra
+                .execute_prepared_coordinator_node(
+                    &prepared_select,
+                    &[ResultValue::Int(key), ResultValue::Int(key)],
+                )
+                .await;
+            assert_eq!(shotover_hit, cassandra_hit);
+        }
+
+        for key in 0..10 {
+            let shotover_hit = shotover
+                .execute_prepared_coordinator_node(
+                    &prepared_update,
+                    &[ResultValue::Int(key), ResultValue::Int(key)],
+                )
+                .await;
+            let cassandra_hit = cassandra
+                .execute_prepared_coordinator_node(
+                    &prepared_update,
+                    &[ResultValue::Int(key), ResultValue::Int(key)],
+                )
+                .await;
+            assert_eq!(shotover_hit, cassandra_hit);
+        }
+
+        for key in 0..10 {
+            let shotover_hit = shotover
+                .execute_prepared_coordinator_node(
+                    &prepared_delete,
+                    &[ResultValue::Int(key), ResultValue::Int(key)],
+                )
+                .await;
+            let cassandra_hit = cassandra
+                .execute_prepared_coordinator_node(
+                    &prepared_delete,
+                    &[ResultValue::Int(key), ResultValue::Int(key)],
+                )
+                .await;
+            assert_eq!(shotover_hit, cassandra_hit);
+        }
+    }
 }
 
-pub async fn create_table(connection: &mut CassandraConnection) {
-    let create_table_cql =
-        "CREATE TABLE IF NOT EXISTS test_routing_ks.my_test_table (key int PRIMARY KEY, name text);";
-    run_query(connection, create_table_cql).await;
+mod composite_key {
+    use crate::helpers::cassandra::{run_query, CassandraConnection, ResultValue};
+
+    async fn create_keyspace(connection: &mut CassandraConnection) {
+        let create_ks: &'static str = "CREATE KEYSPACE IF NOT EXISTS test_routing_ks WITH REPLICATION = { 'class' : 'SimpleStrategy', 'replication_factor' : 1 };";
+        run_query(connection, create_ks).await;
+    }
+
+    async fn create_table(connection: &mut CassandraConnection) {
+        let create_table_cql =
+        "CREATE TABLE IF NOT EXISTS test_routing_ks.my_test_table_composite (key int, name text, age int, blah text, PRIMARY KEY((key, name), age));";
+        run_query(connection, create_table_cql).await;
+    }
+
+    pub async fn test(shotover: &mut CassandraConnection, cassandra: &CassandraConnection) {
```

Comment:
the `shotover` connection doesnt need to be `&mut` as demonstrated by the `cassandra` connection which is just `&`